### PR TITLE
Dtolmaco/add debug log

### PR DESCRIPTION
**Pull request by dantol29**

### CODE REVIEW

File: src/CGIHandler.cpp [hunk 1]
```diff
@@ -157,7 +157,7 @@ bool CGIHandler::executeCGI(const MetaVariables &env, HTTPResponse &response)
 		// execve(argvPointers[0], argvPointers.data(), envpPointers.data());
 		if (execve(argvPointers[0], argvPointers.data(), envpPointers.data()) == -1)
 		{
-			perror("execve");
+			Debug::log("CGIHandler: execve failed", Debug::CGI);
 			return false;
```
Comment (552020) [inline src/CGIHandler.cpp:161]: perror gives extra information and this way we are losing them. Please provide a Debug::log version that integrates the perror functionalities.

File: src/Router.cpp [hunk 2]
```diff
@@ -64,15 +64,14 @@ void Router::routeRequest(HTTPRequest &request, HTTPResponse &response)
...
+	
+	Debug::log("Routing Request: path = " + request.getPath(), Debug::NORMAL);
 
 	PathValidation pathResult = pathIsValid(response, request);
-	std::cout << BLUE << "path: " << request.getPath() << RESET << std::endl;
-	std::cout << BLUE << "PathValidation: " << pathResult << RESET << std::endl;
-	
+	Debug::log("Routing Request: pathResult = " + toString(pathResult), Debug::NORMAL);
+	Debug::log("Path requested: " + request.getPath(), Debug::NORMAL);
 	// check if method is allowed
```
Comment (552020) [inline src/Router.cpp:75]: Should this be Debug::NORMAL or should we have an enum ROUTER?

File: src/Router.cpp
Comment (552020) [inline src/Router.cpp:1]: Add Debug::ROUTER option

File: src/ServerSocket.cpp
Comment (552020) [inline src/ServerSocket.cpp:1]: Why did you remove my logs here?

File: src/StaticContentHandler.cpp
Comment (552020) [inline src/StaticContentHandler.cpp:1]: Is Debug::NORMAL here justified?

File: src/UploadHandler.cpp [hunk 3]
```diff
@@ -78,13 +78,12 @@ bool UploadHandler::checkFiles(const HTTPRequest &request)
 		std::string extension = filename.substr(filename.find_last_of(".") + 1);
 		if (extension.empty())
 		{
-			// TODO: have an html ?
-			std::cout << "415 Unsupported Media Type (Error: file has no extension)" << std::endl;
```
Comment (552020) [inline src/UploadHandler.cpp:82]: Did we solve this TODO?

File: src/UploadHandler.cpp [hunk 4]
```diff
@@ -111,11 +109,11 @@ bool UploadHandler::createFile(HTTPRequest &request)
 		{
 			outfile << it->fileContent;
 			outfile.close();
-			std::cout << "File created successfully at " << filePath << std::endl;
+			Debug::log("File created successfully at " + filePath, Debug::NORMAL);
```
Comment (552020) [inline src/UploadHandler.cpp:112]: Probably all these Debug::NORMAL should be Debug::ERROR and we will always print ERROR (proabably in red). See my http branch for this enhancement. Debug::ERROR could add always RED.

File: src/UploadHandler.cpp [hunk 5]
```diff
@@ -151,13 +148,11 @@ void UploadHandler::handleRequest(HTTPRequest &request, HTTPResponse &response)
...
 	{
-		std::cout << PURPLE << "calling handle response" << RESET << std::endl;
 		handleResponse(response, BAD_REQUEST);
 		return;
 	}
 	if (!request.getUploadBoundary().empty())
 	{
-		std::cout << PURPLE << "calling upload boundary" << RESET << std::endl;
 		if (!createFile(const_cast<HTTPRequest &>(request)))
 			return (response.setStatusCode(422, "Unprocessable Entity"));
```
Comment (552020) [inline src/UploadHandler.cpp:157]: Why remove these two errors log?

**Review by 552020 — CHANGES_REQUESTED** [review]
I commented in the code: but good job!

File: conf/single.conf
[the remark this answers is not in the document]
Comment (dantol29) [reply conf/single.conf:1]: just to check that it works. Also it is a reminder that second listen with different  ip but the same port should work

File: include/webserv.hpp
[the remark this answers is not in the document]
Comment (dantol29) [reply include/webserv.hpp:1]: What do you mean? I just added defines for timeouts

File: src/CGIHandler.cpp [hunk 6]
```diff
@@ -46,18 +45,19 @@ void CGIHandler::handleRequest(HTTPRequest &request, HTTPResponse &response)
 		// TODO: it should be hardcoded
 		response.setBody("500 Internal Server Error");
 	}
```
[the remark this answers is not in the document]
Comment (dantol29) [reply src/CGIHandler.cpp:47]: Dunno. It was added a long time ago. I guess we can remove it now

File: src/CGIHandler.cpp [hunk 7]
```diff
@@ -99,12 +99,12 @@ std::vector<char *> CGIHandler::convertToCStringArray(const std::vector<std::str
...
 {
 	(void)sig;
-	std::cout << "CGIHandler: Timeout" << std::endl;
+	Debug::log("CGIHandler: Timeout", Debug::CGI);
 }
 
 bool CGIHandler::executeCGI(const MetaVariables &env, HTTPResponse &response)
 {
-	std::cout << RED << "Entering CGIHandler::executeCGI" << RESET << std::endl;
+	Debug::log("CGIHandler::executeCGI", Debug::CGI);
```
[the remark this answers is not in the document]
Comment (dantol29) [reply src/CGIHandler.cpp:107]: Lets do it in your PR, when we gonna merge this one

File: src/CGIHandler.cpp (hunk 1 above)
Comment (dantol29) [reply src/CGIHandler.cpp:161]: Alright

File: src/Router.cpp
Comment (dantol29) [reply src/Router.cpp:1]: Coming soon in the next PR

File: src/Router.cpp (hunk 2 above)
Comment (dantol29) [reply src/Router.cpp:75]: Of course ROUTER

File: src/ServerSocket.cpp
Comment (dantol29) [reply src/ServerSocket.cpp:1]: Because we added them when it was not working properly, now it works fine

File: src/UploadHandler.cpp (hunk 3 above)
Comment (dantol29) [reply src/UploadHandler.cpp:82]: We do not need a separate HTML for this

File: src/UploadHandler.cpp (hunk 4 above)
Comment (dantol29) [reply src/UploadHandler.cpp:112]: Then gonna do it in your PR

File: src/UploadHandler.cpp (hunk 5 above)
Comment (dantol29) [reply src/UploadHandler.cpp:157]: It is not necessary because handleResponse and createFIle have prints inside

**Comment by dantol29** [comment]
Currently working on 503(Service Unavailable):


NGINX allows you to manage the number of simultaneous client connections using the limit_conn module. When the connection limit is exceeded, NGINX will return a 503 Service Unavailable status code, helping to protect your server from being overwhelmed by too many concurrent connections

**Comment by dantol29** [comment]
503 finished! I wrote explanation here https://linear.app/web-serv/issue/WEB-340/503-service-unavailable-too-many-requests

**Review by lmangall — APPROVED** [review]
LGTM !

**Review by 552020 — APPROVED** [review]
LPGTM
